clouddriver-aws/src/test/groovy/com/netflix/spinnaker/clouddriver/aws/provider/agent/AmazonInstanceTypeCachingAgentSpec.groovy

```diff
@@ -111,4 +111,31 @@ class AmazonInstanceTypeCachingAgentSpec extends Specification {
     0 * _
   }
 
+  void "should sort all by instance type then size"() {
+    when:
+    def result = agent.loadData(providerCache)
+
+    then:
+    1 * ec2.describeReservedInstancesOfferings(new DescribeReservedInstancesOfferingsRequest()) >> new DescribeReservedInstancesOfferingsResult(
+      reservedInstancesOfferings: [
```

Comment:
Might be worth throwing a couple extra test scenarios:
- an invalid instance type  (ie. `m3.supersmall`)
- xl,2xl, 4xl, 8xl, 16xl